### ISSUE TITLE
[enhancement] priority_sel should support any cases type

### ISSUE THREAD

**Issue by proppy** [google/xls#1729]
### What's hard to do? (limit 100 words)

currently `priority_sel` can only be used w/ scalar type (`uN`), see: 
https://google.github.io/xls/dslx_std/#priority_sel

### Current best alternative workaround (limit 100 words)

using  `one_hot` in combination w/ a `match` expression decoding the one_hot result and cases w/ arbitrary expression allow support for compound type like `struct`, `tuple` and `array`.

### Your view of the "best case XLS enhancement" (limit 100 words)

making `priority_sel` support arbitrary types in the case array.

**Comment by ericastor** [comment]
This is a good idea, but it's worth noting that using a `match` expression will currently work just fine - and as it happens, it compiles down to a `priority_sel`.

**Comment by proppy** [comment]
> it's worth noting that using a match expression will currently work just fine for arbitrary types 

Yes, I re-discovered this pattern which allow you to encode as `priority_sel` w/ arbitrary type as a `match` statement w/ exclusive condition:

```
match true {
   condition0 => result0,
   condition1 => result1,
   condition2 => result2,
   condition3 => result3,
}
```

**Comment by proppy** [comment]
> I re-discovered this pattern

Note this is not actually working as stated above, see #1909
